cocotbext/axi/axis.py

```diff
@@ -302,7 +303,7 @@ def __init__(self, bus, clock, reset=None, reset_active_level=True,
             if hasattr(self.bus, sig):
                 if self._init_x and sig not in ("tvalid", "tready"):
                     v = getattr(self.bus, sig).value
-                    v.binstr = 'x'*len(v)
+                    v = apply_binstr(v, 'x'*len(v))
```

Comment:
I am observing another change with latest cocotb HEAD. 2.0.0.dev0+8f9456fc

```
File "/home/mto/github_cocotb/cocotbext-axi/cocotbext/axi/axis.py", line 305, in __init__
v.binstr = 'x'*len(v)
TypeError: object of type 'Logic' has no len()
```

As far as i can see this updated code still uses len() which i guess will not work. Unfortunately i don't know what needs to be used instead.. Also it might affect more lines elsewhere

Reply:
Added a length operator to `Logic` for backwards compat. https://github.com/cocotb/cocotb/pull/4785

Going forward this library will need to do a better job of handling arrays vs scalar arguments.